sfcasts/factory-families.md

```diff
@@ -1,86 +1,105 @@
-Ok, finish this up by adding a `default` case and print an `Invalid Cheat Code` message.
-
-Perfect! Before giving this a try, there's a tiny detail we need to handle. Symfony
-does not know what `AttackTypeFactory` to inject into `CharacterBuilder` because we 
-have more than one implementation of the `AttackTypeFactoryInterface` so we need to
-tell Symfony which one to use by default. To do this I'll leverage the `AsAlias` attribute,
-open up `AttackTypeFactory` and above the class name write `#[AsAlias()]` and pass
-`AttackTypeFactoryInterface::class` as the id.
-
-Ok, finally! Let's give this a try. Spin over to your terminal and run:
+All right! We're getting closer! Now we need a way to *play* our cheat codes.
+We'll handle them as command-line options, so open up `GameCommand` and,
+below the constructor, write `protected function configure()`. Inside, add a new
+option by calling `$this->addOption()`. The *first* argument is the option's
+*name*. We'll call it `cheatCode`. The *second* argument is the *shortcut*.
+Let's use `c`. The *third* argument is going to be an *optional* value, so let's
+set it to `InputOption::VALUE_OPTIONAL`. Then, inside the `execute()` method,
+before selecting the character, we'll check to see if the `cheatCode` option was
+passed in, and if *so*, we'll activate it.
+
+To do this, write `if ($input->getOption('cheatCode'))`, and inside *that*,
+`$this->game->activateCheatCode()`, sending the `cheatCode` option as the
+argument. This method doesn't exist yet, so let's create it. Position the cursor
+over the method's name, hold "Option" + "Enter", and select "Add method". Okay,
+change the argument to `string $cheatCode`... and inside, we'll use a
+`switch-case` just in case we want to add more cheat codes in the future. To do
+that, say `switch ($cheatCode)` and inside that, we'll add a `case` with the
+value of our *ultimate* cheat code. Hm... what would be a good value for that?
+Oh! I know! I'll paste this in because it's a bit long, but it *might* look
+familiar to you. Remember the famous Konami code? It feels like the 90s are
+back!
+
+Okay, *inside* the `case`, let's print a message so we know that the cheat code
+was activated. *Then* we'll swap the factory on the `CharacterBuilder`. To do
+that, write
+`$this->characterBuilder->setAttackTypeFactory(new UltimateAttackTypeFactory())`
+and add a `break` at the end. *Awesome*.
+
+Now, you may be thinking "What if the `UltimateAttackTypeFactory` has
+dependencies?" or "What if it's not that simple to instantiate?", and that's a
+valid concern. A way to solve that is by doing the same thing we did with the
+"state" classes - by leveraging the [`AutowireLocator` attribute](https://bit.ly/sf-service-locator-attribute).
+Another option would be to create a factory *for* your factories.
+Ohh factory-ception! I sure hope Skynet isn't listening... Ok, we can finish this
```

Comment:
@kbond I think this is a good joke but I don't know how natural it may sound. Feel free to change it if you want